### PR TITLE
ENG-12663: Add authentication timeout to the Java client.

### PR DESCRIPTION
**Pull request by nshi**
Time out authentication on the client side if it takes longer than the
connection timeout setting in case the server dies or the connection is broken
during the authentication process. It is governed by the same connection timeout
setting specified in the ClientConfig.

If authentication times out, it throws an IOException.

### CODE REVIEW

File: src/frontend/org/voltdb/client/ConnectionUtil.java [hunk 1]
```diff
@@ -94,6 +98,10 @@ private void shutdown() throws InterruptedException {
 
     private static final GSSManager m_gssManager = GSSManager.getInstance();
 
+    // Thread pool for checking authentication timeout
+    private static final ScheduledThreadPoolExecutor m_periodicWorkThread =
+        CoreUtils.getScheduledThreadPoolExecutor("Periodic Work", 0, CoreUtils.SMALL_STACK_SIZE);
```
Comment (akhanzode) [inline src/frontend/org/voltdb/client/ConnectionUtil.java]: Can you call this "Authentication Timer" or something unless we intend to use this for other tasks.

File: src/frontend/org/voltdb/client/ConnectionUtil.java [hunk 2]
```diff
@@ -171,20 +180,38 @@ public DelegatePrincipal apply(Principal input) {
...
+        // Setup a timer that times out the authentication if it is stuck (server dies, connection drops, etc.)
+        final ScheduledFuture<?> timeoutFuture;
+        if (timeoutMillis > 0) {
+            timeoutFuture = m_periodicWorkThread.schedule(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        aChannel.close();
+                    } catch (IOException e) {
+                        // Ignore
```
Comment (akhanzode) [inline src/frontend/org/voltdb/client/ConnectionUtil.java]: For client this may be useful to log otherwise we wont know if client is timing out.

Comment (nshi) [inline src/frontend/org/voltdb/client/ConnectionUtil.java]: We don't log in the client library.

It throws an IOException if authentication timed out. User would know.

File: src/frontend/org/voltdb/client/ConnectionUtil.java (hunk 1 above)
Comment (nshi) [inline src/frontend/org/voltdb/client/ConnectionUtil.java]: Will do.

**Pull request merged**